pages/api/text.ts

```diff
@@ -0,0 +1,62 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { gql, OperationResult } from 'urql'
+import { fetchData, getEns } from './_utils'
+
+const ProfileTextRecordsQuery = gql<{
+  domains: [{ resolver: { texts: string[] } }]
+}>`
+  query ProfileTextRecords($name: String!) {
+    domains(where: { name: $name }) {
+      resolver {
+        texts
+      }
+    }
+  }
+`
+
+export async function getEnsTextRecords(
+  ensName: string
+): Promise<{ key: string; value: string }[]> {
+  // 1. Check which text records are set for the domain with ENS subgraph,
+  // to prevent unnecessary contract calls.
+  const result: OperationResult<{
+    domains: [{ resolver: { texts: string[] } }]
+  }> = await fetchData(
+    ProfileTextRecordsQuery,
+    { name: ensName },
+    {
+      url: `https://api.thegraph.com/subgraphs/name/ensdomains/ens`,
+      requestPolicy: 'cache-and-network'
+    }
+  )
+  if (!result?.data?.domains[0]?.resolver) throw 'No ENS text records found'
```

Comment:
Note for later when I have figured out Vercel deployment of this project: we need to make sure if nothing is found or set, we do not error out. Nothing found should be `200` repsonse too, as request and response actually works. So rather we probably should have a specific response if nothing is found

Reply:
Yeah, I've adjusted this so that 200 is always sent whether or not something is found. This is also part of the tests